### PR TITLE
Porting interface module

### PR DESCRIPTION
**Pull request by surajkannur**
Porting Interface module 

## Description

The interface module has been ported to pure python as part of shifting to use nanobind

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/main/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [x] I have updated the [credits](https://github.com/glotzerlab/freud/blob/main/doc/source/reference/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/main/ChangeLog.md).

### CODE REVIEW

**Comment by surajkannur** [comment]
pre-commit.ci autofix

**Comment by surajkannur** [comment]
pre-commit.ci autofix

File: freud/interface.py [hunk 1]
```diff
@@ -47,15 +44,9 @@ def compute(self, system, query_points, neighbors=None):
...
-            freud.locality.NeighborQuery nq
-            freud.locality.NeighborList nlist
-            freud.locality._QueryArgs qargs
-            const float[:, ::1] l_query_points
-            unsigned int num_query_points
 
-        nlist = freud.locality._make_default_nlist(
-            system, neighbors, query_points)
+        # freud.locality.
+        nlist = _make_default_nlist(system, neighbors, query_points)
```
Comment (DomFijan) [inline freud/interface.py]:
```suggestion
        nlist = _make_default_nlist(system, neighbors, query_points)
```
That comment was not doing anything.

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**